src/EnergyPlus/DataSurfaces.hh

```diff
@@ -150,6 +150,19 @@ namespace DataSurfaces {
         Num
     }; // Valid window shading types: IntShade <= Type <= BGBlind; the rest are shading status
 
+    constexpr std::array<std::string_view, static_cast<int>(WinShadingType::Num)> WindowShadingTypeNames = {
```

Comment:
Nice use of enums and arrays.

Reply:
Move these to the `.cc` file and add `extern const` declaration here if necessary (if they are used anywhere outside this module). [This is a new pattern, I only learned about the downsides of using `constexpr` on non-scalars in `.hh` files recently.]

Reply:
When I tried this with VS 2019 Version 16.11.9 I got a compiler error.

https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2370?view=msvc-170